src/isomedia/isom_write.c

```diff
@@ -4434,6 +4435,11 @@ GF_Err gf_isom_clone_track(GF_ISOFile *orig_file, u32 orig_track, GF_ISOFile *de
 	if (dest_file->moov->mvhd->nextTrackID <= new_tk->Header->trackID)
 		dest_file->moov->mvhd->nextTrackID = new_tk->Header->trackID+1;
 
+        /*if it contains IAMF, add the iamf brand to ftyp*/
+        if (gf_isom_get_media_subtype(dest_file, new_tk->Header->trackID, 1) == GF_ISOM_BOX_TYPE_IAMF) {
```

Comment:
This compares a subtype and a box type.

Reply:
Thanks, I missed this. Looks like I could just iterate through the SampleDescription's child boxes and check their type.

Reply:
No, you just need to declare a subtype for IAMF in isomedia.h (with same 4CC as the internal box type)

Reply:
Done